py/server/deephaven/table.py

```diff
@@ -343,6 +403,65 @@ def with_filters(self, filters: Union[str, Filter, Sequence[str], Sequence[Filte
 
+    def layout_hints(self, front: Union[str, List[str]] = None, back: Union[str, List[str]] = None,
+                     freeze: Union[str, List[str]] = None, hide: Union[str, List[str]] = None,
+                     column_groups: List[dict] = None, search_display_mode: SearchDisplayMode = None) -> Table:
+        """ Sets layout hints on the TreeTable
+
+        Args:
+            front (Union[str, List[str]]): the columns to show at the front.
+            back (Union[str, List[str]]): the columns to show at the back.
+            freeze (Union[str, List[str]]): the columns to freeze to the front.
+                These will not be affected by horizontal scrolling.
+            hide (Union[str, List[str]]): the columns to hide.
+            column_groups (List[Dict]): A list of dicts specifying which columns should be grouped in the UI.
+                The dicts can specify the following:
+
+                * name (str): The group name
+                * children (List[str]): The column names in the group
+                * color (Optional[str]): The hex color string or Deephaven color name
+            search_display_mode (SearchDisplayMode): set the search bar to explicitly be accessible or inaccessible,
+                or use the system default. :attr:`SearchDisplayMode.SHOW` will show the search bar,
+                :attr:`SearchDisplayMode.HIDE` will hide the search bar, and :attr:`SearchDisplayMode.DEFAULT` will
+                use the default value configured by the user and system settings.
+
+        Returns:
+            a new tree table with the layout hints set
+
+        Raises:
+            DHError
+        """
+        try:
+            _j_layout_hint_builder = _JLayoutHintBuilder.get()
+
+            if front is not None:
+                _j_layout_hint_builder.atFront(to_sequence(front))
+
+            if back is not None:
+                _j_layout_hint_builder.atBack(to_sequence(back))
+
+            if freeze is not None:
+                _j_layout_hint_builder.freeze(to_sequence(freeze))
+
+            if hide is not None:
+                _j_layout_hint_builder.hide(to_sequence(hide))
+
+            if column_groups is not None:
+                for group in column_groups:
+                    _j_layout_hint_builder.columnGroup(group.get("name"), j_array_list(group.get("children")),
+                                                       group.get("color", ""))
+
+            if search_display_mode is not None:
+                _j_layout_hint_builder.setSearchBarAccess(search_display_mode.value)
+
+        except Exception as e:
+            raise DHError(e, "failed to create layout hints") from e
+
+        try:
+            return TreeTable(j_tree_table=self.j_tree_table.setLayoutHints(_j_layout_hint_builder.build()),
+                             id_col=self.id_col, parent_col=self.parent_col)
+        except Exception as e:
+            raise DHError(e, "failed to set layout hints on tree table") from e
 
```

Comment:
Should `PartitionedTable` or `PartitionedTableProxy` also have this method?

Reply:
Those classes don't extend `GridAttributes` which has the `setLayoutHints` method defined in Java.